tests/backuprestore/backup_restore_suite_test.go

```diff
@@ -0,0 +1,140 @@
+// TEST_LABEL_FILTER=backup-restore  /usr/local/go/bin/go test -timeout 60m github.com/rancher/observability-e2e/tests/bacup-restore -v -count=1 -ginkgo.v
+func TestE2E(t *testing.T) {
+	RegisterFailHandler(FailWithReport)
+	suiteConfig, reporterConfig := GinkgoConfiguration()
+
+	// Set the label filter to "LEVEL0" (or any other test with custom tag)
+	if envLabelFilter := os.Getenv("TEST_LABEL_FILTER"); envLabelFilter != "" {
+		suiteConfig.LabelFilter = envLabelFilter
+	} else {
+		suiteConfig.LabelFilter = "LEVEL0"
+	}
+	logger.Logf("Executing tests with label '%v'", suiteConfig.LabelFilter)
+	RunSpecs(t, "Backup and Restore End-To-End Test Suite", suiteConfig, reporterConfig)
+}
+
+// This setup will run once for the entire test suite
+var _ = BeforeSuite(func() {
+	project = nil
+
+	testSession := session.NewSession()
+	sess = testSession
+	client, err = rancher.NewClient("", testSession)
+	Expect(err).NotTo(HaveOccurred(), "Failed to create Rancher client: %v", err)
+
+	// Get clusterName from config yaml
+	clusterName := client.RancherConfig.ClusterName
+	Expect(clusterName).NotTo(BeEmpty(), "Cluster name to install is not set")
+
+	// Get cluster meta
+	cluster, err = clusters.NewClusterMeta(client, clusterName)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Get Server and Registry Setting Values
+	registrySetting, err = client.Management.Setting.ByID("system-default-registry")
+	Expect(err).NotTo(HaveOccurred())
+
+	projectsList, err := client.Management.Project.List(&types.ListOpts{
+		Filters: map[string]interface{}{
+			"clusterId": cluster.ID,
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	for i := range projectsList.Data {
+		p := &projectsList.Data[i]
+		if p.Name == exampleAppProjectName {
+			project = p
+			break
+		}
+	}
+
+	// Check if project was found
+	if project == nil {
+		projectConfig := &management.Project{
+			ClusterID: cluster.ID,
+			Name:      exampleAppProjectName,
+		}
+
+		project, err = client.Management.Project.Create(projectConfig)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(project.Name).To(Equal(exampleAppProjectName))
+	}
+
+	// Load configuration from the default YAML file
+	backupRestoreConfig = &localConfig.BackupRestoreConfig{}
+	err = utils.LoadConfigIntoStruct(charts.BackupRestoreConfigurationFileKey, backupRestoreConfig)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Initialize the S3 client
+	s3Client, err = resources.NewS3Client(backupRestoreConfig)
+	Expect(err).NotTo(HaveOccurred())
+
+	// Create the S3 bucket from the config
+	err = s3Client.CreateBucket(backupRestoreConfig.S3BucketName, backupRestoreConfig.S3Region)
+	Expect(err).NotTo(HaveOccurred())
+	logger.Logf("S3 bucket '%s' created successfully", backupRestoreConfig.S3BucketName)
+
+})
+
+// This teardown will run once after all the tests in the suite are done
+var _ = AfterSuite(func() {
+	// Delete the S3 bucket
+	err := s3Client.DeleteBucket(backupRestoreConfig.S3BucketName)
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
if incase bucket is not created because of failure
```suggestion
        if s3Client != nil {
	    err := s3Client.DeleteBucket(backupRestoreConfig.S3BucketName)
	    Expect(err).NotTo(HaveOccurred())
	}
```

Reply:
this check is not required as it is already covered in the BeforeSuite.